### PR TITLE
Feature | Thoughts

### LINKED ISSUE #249: Bounty 6$ | Feature | Thought
### What ?
- At the moment, users can write blogs. Blogs are long form content. 
- Introducing Thoughts. User can write a thought. Thoughts are short form. Think of 1 Thought like a Twitter post or Threads post. 

### How ?

1. backend/prisma/schema.prisma - 

```javascript
model Thought {
  id            String     @id @default(uuid())
  content       String
  publishedDate DateTime   @default(now())
  published     Boolean    @default(false)
  author        User       @relation(fields: [authorId], references: [id], onDelete: Cascade)
  authorId      String
}
```
Run prisma migrations to update db with schema

2. backend/src/routes/thought.ts - thoughtRouter
```bash
thoughtRouter.post
thoughtRouter.put
thoughtRouter.get: 1 and all
thoughtRouter.delete
```
create, update, delete - protected routes
get - public routes

3. Test all routes on Postman (optional)

<hr />

4. frontend/src/pages/Thought.ts - 
```bash
1. Create Thought button 
2. Show all Thoughts
3. Delete Thought 
4. Update Thought
```

The frontend should be simple, page/Thought.ts is a temporary placeholder to test the backend setup. We will eventually integrate it into the App.

### PR DESCRIPTION
**Pull request by chiraggowda92187**
# Pull Request Title

## Description

Resolves #[Issue Number]

## Linked Issues

- Fixes #249 
- Closes #249

## Type of Change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Documentation update

## Changes

- Change 1
- Change 2
- ...

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I assure there are no similar/duplicate pull requests regarding the same issue
- [x] My changes follow the project's coding guidelines and best practices
- [x] Code is formatted properly and lint check passes successfully
- [x] I have made corresponding changes to the documentation (if applicable)

### CODE REVIEW

File: frontend/src/config.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
-export const BACKEND_URL = 'https://backend.aadesh-kulkarni08.workers.dev';
+// export const BACKEND_URL = 'https://backend.aadesh-kulkarni08.workers.dev';
+export const BACKEND_URL = 'http://localhost:39013';
```
Comment (aadeshkulkarni) [inline frontend/src/config.ts:2]: Please rollback this change.

File: frontend/src/pages/Thought.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                            </div>
+                        })}
+                    </div>
+                </div>
+
+
+            </div>
+        </div>
+    )
+}
```
Comment (aadeshkulkarni) [inline frontend/src/pages/Thought.tsx:61]: The CRUD APIs should be functional on the frontend. 
User should be able to perform create, read, update & delete from the UI. 
At the moment, that's not happening.

**Review by aadeshkulkarni — CHANGES_REQUESTED** [review]
Vercel build fails with these errors:

Screenshot 2024-07-21 at 12 21 09 PM

The @aadeshk/medium-common errors will only be fixed after re-publishing the npm package. But, the rest of errors should be fixable.

**Comment by chiraggowda92187** [comment]
I have made the changes and added features as requested. Could you review it sir.? 
Thank you.

File: frontend/src/config.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 export const BACKEND_URL = 'https://backend.aadesh-kulkarni08.workers.dev';
+// export const BACKEND_URL = 'http://localhost:39013';
```
Comment (aadeshkulkarni) [inline frontend/src/config.ts:2]: please remove this line

File: backend/.gitignore [hunk 4]
```diff
@@ -9,4 +9,5 @@ yarn.lock
 pnpm-lock.yaml
 bun.lockb
 .env
-wrangler.toml
\ No newline at end of file
+wrangler.toml
+wrangler.sample.toml
```
Comment (aadeshkulkarni) [inline backend/.gitignore:13]: No need of adding this in .gitignore

Comment (chiraggowda92187) [reply backend/.gitignore]: wrangler.toml was by default added in the original repo I removed and added it by mistake git detected it as a addition. wrangler.sample.toml was added as a extra I have removed it. Also removed the commented out url.

**Pull request closed without merging**